cradle-core/src/test/java/com/exactpro/cradle/BookInfoTest.java

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright 2024 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exactpro.cradle;
+
+import com.google.common.collect.Lists;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import javax.annotation.Nullable;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Random;
+
+import static com.exactpro.cradle.Order.DIRECT;
+import static com.exactpro.cradle.Order.REVERSE;
+import static com.google.common.collect.Lists.newArrayList;
+import static java.lang.Math.min;
+import static java.time.temporal.ChronoUnit.DAYS;
+import static java.time.temporal.ChronoUnit.NANOS;
+import static java.util.Collections.emptyList;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertSame;
+
+public class BookInfoTest {
+    public static Random RANDOM = new Random();
+    public static final BookId BOOK_ID = new BookId("test-book");
+    private static final List<PageInfo> PAGES;
+
+    static {
+        List<PageInfo> pages = new ArrayList<>();
+        Instant start = Instant.now().minus(7, ChronoUnit.DAYS);
+        Instant end = Instant.now();
+        Instant current = start;
+        Instant previous;
+        do {
+            previous = current;
+            current = current.plus(1, ChronoUnit.HOURS);
+            pages.add(createPageInfo(previous, current));
+        } while (current.isBefore(end));
+        pages.add(createPageInfo(current, null));
+
+        PAGES = Collections.unmodifiableList(pages);
+    }
+
+    @Test(dataProvider = "cacheSize")
+    public void lazyPageAddTest(int cacheSize) {
```

Comment:
Just a small note: what did we break if this test fails? I tried to figure that out for some time and the only answer I had was 'something with pages'. In my personal opinion, unit tests should be as concise as possible and answer the question: what did we break? And this test does not answer this question.
This is not a request to change it immediately, just a piece of thought about what unit tests should look like.

Reply:
This test was made as hot check. I will split it to several part at the end of task